### ISSUE TITLE
ChineseCalendar shows nothing but black window

### ISSUE THREAD

**Issue by retsyo** [XusinboyBekchanov/VisualFBEditor#583]
lastet VisualFBEditor and myfbframework, but ChineseCalendar shows nothing but black window
中文：使用刚下载的VisualFBEditor和myfbframework，但是编译出来的ChineseCalendar窗口一片黑，其余啥都没有

![buggy_clock](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/7960913/42c8ffa9-2b59-44e9-a572-178e8087faa9)

**Comment by chunmingwang** [comment]
I haven't had the time to download the latest VFBE and MFF yet. I'll try them out when I have some free time.
我还没有空下载最新的vfbe和mff，有空去试试看

**Comment by retsyo** [comment]
the buggy one is the ChineseCalendar in VisualFBEditor, not that one in your repositories
 
是 VisualFBEditor 里面的例子只有黑色，不是你代码库中的例子

**Comment by chunmingwang** [comment]
it‘s same to me, maybe this PR caused it
和我一样，也许是这个pr导致的
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/d7821367f1e80fffd97179448a68c9dfbd5060f7

**Comment by XusinboyBekchanov** [comment]
Yes, after hustbeef's pull request this happened. 

After this fix it now shows:

Update DrawClockCalendar.bas:
https://github.com/XusinboyBekchanov/VisualFBEditor/commit/1faa6d3ebb1c9e2c8e74133cc195fca7280754fc

**Comment by hustbeef** [comment]
Do you updated the MFF? It is fine working in my laptop.  
`Canvas.Pen.Color = mClr(0)
	'Canvas.Line 0, 0, Canvas.Width, Canvas.Height, mClr(0) , "F"` this function is like backcolor but it is very slow. So I remove this code and add  code in frmClock.frm `	' Panel1
		With Panel1
			.Name = "Panel1"
                          ......
			'.BackColor = -1
			.DoubleBuffered = True
			**.BackColor = 16766975**
			.Transparent = True
			...........
		End With`
_It is better remove this code and make everthing is fine.

**Comment by hustbeef** [comment]
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/a144c2e5-d5da-42f6-8c0c-75f3c0509a40)

**Comment by XusinboyBekchanov** [comment]
There BackColor is set to -1, so it will appear black.

**Comment by hustbeef** [comment]
Recently I upgrade the control Panel , Picture and canvas for better drawing. I think It's too simple to draw a more realistic hour, minute, and second hand now.

**Comment by chunmingwang** [comment]
Usually, the painting is first drawn on the memory DC, and then copied and overlaid in the form DC for display. Avoid flickering caused by drawing and display at the same time,this is call double buffer.

通常绘画都是先绘制在内存设备上，完成后再复制覆盖在窗口显示出来
避免一面绘制一面显示导致的画面闪烁，这就是双缓存.

**Comment by hustbeef** [comment]
Yes. Better set .DoubleBuffered = True 
It isI forgot updated the frmClock.frm with .BackColor =ClPink. Switch the mode on menu Item "Clock"
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/e78e9730-19a9-4f79-ab48-4cf1243d7947)
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/dbf22e2b-969e-40d5-add2-c8458d02acf4)

**Comment by chunmingwang** [comment]
Analog clocks require better window effects, and it is better to use the UpdateLayeredWindow.
You can refer to UEZ's GDI+Swiss Railway Clock

指针时钟需要更好的效果最好用UpdateLayeredWindow窗口
可以参考UEZ的GDI+ Swiss Railway Clock
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/35757455/8ccc1461-5b6b-4afd-a2d5-378228b79577)

https://www.freebasic.net/forum/viewtopic.php?t=26454

**Comment by hustbeef** [comment]
Because I am drawing with a low-quality PNG file. Now I have improved the code, fixed drawing on forms directly, and removed the function SetLayeredWindowAttributes, It could be working on OS Linux in the future.

![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/ce0635f4-d6fb-41e3-8e27-0b57204e46c1)

![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/508536d7-9876-4ac4-b8a1-18b6d6922b1c)

**Comment by chunmingwang** [comment]
![gdip_click](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/35757455/e449494b-2637-4ba0-a72a-86b79dcaae30)

**Comment by XusinboyBekchanov** [comment]
A very good example, you can update ChineseCalendar.

**Comment by chunmingwang** [comment]
Yes, I will.
I am still fine-tuning some issues.

**Comment by hustbeef** [comment]
I spend 3 hours to upgrade the png files. So I post it.
![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/1d83b770-142e-4a27-9818-506ac7519b9e)

**Comment by chunmingwang** [comment]
I release a PR to add a gdipClock example
https://github.com/XusinboyBekchanov/VisualFBEditor/pull/586

**Comment by retsyo** [comment]
Now ChineseCalendar shows a traditional round clock on running. The menu item, for example, "blink colon" has nothing to do with it. In other words, this has become a new project. Furthur I met bugs, for example clock can only be clicked through even I did not choose "click through". I suggest open a new demo instead of overwritting the old one.

I also suggest to modify the code in own fork. Only do pull requests if new code has been tested throughly on different OS, for example different Windows and Linux in a virtualization software.

Although there is not so many coders who can really contribute to this project, I think stability is always a good consideration.
 
-------
现在的ChineseCalendar一运行就显示一个传统钟表，和菜单里面“不显示秒”等等根本无关。换言之，这完全是另外一个项目；而且有bug（没选择“click through”，不知咋的，钟表就只能“click through”了）。建议另建成一个新的示例，而不要覆盖之前的

官方代码，建议自己fork，在多个平台（例如，虚拟机里跑不同windows、linux版本）实验确定没问题，再申请合并回官方代码

尽管能做贡献的开发人员少，但是也要向着稳定的方向开发啊

**Comment by chunmingwang** [comment]
The example of the Chinese calendar still retains a text-only version of the clock in my repositories, where you can go back to normal and use it.

中国日历这个例子，还保留着一个纯文字时钟的版本在我的程序库里，可以去那里恢复正常的来使用

https://github.com/chunmingwang/ChineseCalendar

**Comment by hustbeef** [comment]
> Now ChineseCalendar shows a traditional round clock on running. The menu item, for example, "blink colon" has nothing to do with it. In other words, this has become a new project. Furthur I met bugs, for example clock can only be clicked through even I did not choose "click through". I suggest open a new demo instead of overwritting the old one.
> 
> I also suggest to modify the code in own fork. Only do pull requests if new code has been tested throughly on different OS, for example different Windows and Linux in a virtualization software.
> 
> Although there is not so many coders who can really contribute to this project, I think stability is always a good consideration.
> 
> 现在的ChineseCalendar一运行就显示一个传统钟表，和菜单里面“不显示秒”等等根本无关。换言之，这完全是另外一个项目；而且有bug（没选择“click through”，不知咋的，钟表就只能“click through”了）。建议另建成一个新的示例，而不要覆盖之前的
> 
> 官方代码，建议自己fork，在多个平台（例如，虚拟机里跑不同windows、linux版本）实验确定没问题，再申请合并回官方代码
> 
> 尽管能做贡献的开发人员少，但是也要向着稳定的方向开发啊

You are correct. We need to address the bugs. The new "Clock" example, demonstrating how to draw an image with canvas, will be relocated to "Controls\MyFbFramework\examples\".

**Comment by chunmingwang** [comment]
Through the development progress of examples, it also helps to promote the progress of mff.

**Comment by XusinboyBekchanov** [comment]
>You are correct. We need to address the bugs. The new "Clock" example, demonstrating how to draw an image with canvas, will be relocated to "Controls\MyFbFramework\examples".

You can send a pull request. Only the name of this example should be slightly different.


>Through the development progress of examples, it also helps to promote the progress of mff.

You're right.

**Comment by hustbeef** [comment]
The new Transparent mode shows the background. I just tried and got it. The code is still not ready. The logic bothers me a lot. But the new Canvas is drawing smooth and fast, that's the goal. Please try running the example code CanvasDraw.bas and resizing the forms by mouse to improve the code. 
Everyone could be write code and improve it. That's the best practice.

**Comment by hustbeef** [comment]
Fixed: Add Transparent of LED Digital showing (PR #587)

![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/52040833/1fc5d1d7-c5ec-4f2e-98a2-9d2c1cc044e6)

**Comment by chunmingwang** [comment]
> Fixed: Add Transparent of LED Digital showing (PR #587)
> 

FX-LED text clock in Example of gdipClock:

![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/35757455/b3bd2447-1d03-4768-8eca-820fcf52b661)


> Now ChineseCalendar shows a traditional round clock on running. The menu item, for example, "blink colon" has nothing to do with it. In other words, this has become a new project. Furthur I met bugs, for example clock can only be clicked through even I did not choose "click through". I suggest open a new demo instead of overwritting the old one.

Analog clock in Example of gdipClock:

![image](https://github.com/XusinboyBekchanov/VisualFBEditor/assets/35757455/098a0cc5-1a0e-4a19-894c-c3664d727640)

**Comment by chunmingwang** [comment]
Should I submit a PR to restore ChineseCalendar as a pure text clock?

**Comment by retsyo** [comment]
> Should I submit a PR to restore ChineseCalendar as a pure text clock?

I think it is a good idea

**Comment by chunmingwang** [comment]
https://github.com/XusinboyBekchanov/VisualFBEditor/pull/600
Restore ChineseCalendar as a pure text clock
Update lunar calender caculate to year 2199

**Comment by hustbeef** [comment]
> Should I submit a PR to restore ChineseCalendar as a pure text clock?
Yes. It is OK. The new "Clock" example, demonstrating how to draw an image with canvas, will be relocated to "Controls\MyFbFramework\examples". And it is can be run under OS Linux also.

**Comment by chunmingwang** [comment]
> > Should I submit a PR to restore ChineseCalendar as a pure text clock?
> 
> I think it is a good idea

i update gdipClock Example, u can try it. leave your message, if u like it or if u haves uggestion for it
